lab_1_classify_by_unigrams/main.py

```diff
@@ -34,6 +44,14 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not (isinstance(tokens, list) and all(isinstance(i, str) for i in tokens)):
+        return None
+
+    freq_dict = {}
+    for letter in tokens:
+        if letter not in freq_dict:
+            freq_dict[letter] = tokens.count(letter) / len(tokens)
```

Comment:
You calculate the same value many times for similar symbols. Can you optimize it and calculate it only for unique